pkg/controller/healthcheck/general/daemonset.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package general
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extensions/pkg/controller/healthcheck"
+
+	"github.com/gardener/gardener/pkg/utils/kubernetes/health"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// DaemonSetHealthChecker contains all the information for the DaemonSet HealthCheck
+type DaemonSetHealthChecker struct {
+	logger         logr.Logger
+	seedClient     client.Client
+	shootClient    client.Client
+	deploymentName string
+}
+
+// CheckStatefulSet is a healthCheck function to check DaemonSets
+func CheckDaemonSet(name string) healthcheck.HealthCheck {
```

Comment:
```suggestion
func NewDaemonSetHealthChecker(name string) healthcheck.HealthCheck {
```

`CheckDaemonSet` sounds like this func is doing the action. But it is only returning new instance of the checker.